packages/gafl-webapp-service/src/pages/summary/contact-summary/__tests__/route.spec.js

```diff
@@ -33,349 +36,274 @@ const route = require('../route.js').default
+  ])('changeLicenceDetails should be %s when isLicenceForYou is %s', async (mssg, isLicenceForYou, mssgKey) => {
+    const permission = getMockPermission(
+      {},
+      {
+        licenceLength: '1D',
+        isLicenceForYou
+      }
+    )
+    const mssgCatalog = getMockCatalog({
+      [mssgKey]: mssg
     })
-
-    describe('when purchasing a 12 month (physical licence) without postal fulfilment', () => {
-      it('should display the Licence and Contact as the email and Newsletter as no', async () => {
-        const permission = {
-          licenceLength: '12M',
-          licensee: {
-            ...addressAndContact,
-            postalFulfilment: false,
-            preferredMethodOfConfirmation: 'Email',
-            preferredMethodOfReminder: 'Email',
-            email: 'new3@example.com',
-            preferredMethodOfNewsletter: 'Prefer not to be contacted'
-          }
-        }
-        const sampleRequest = generateRequestMock(permission)
-
-        const { summaryTable } = await getData(sampleRequest)
-
-        expect(summaryTable).toMatchSnapshot()
-      })
-
-      it('should display the Licence and Contact as the phone number and Newsletter as yes', async () => {
-        const permission = {
-          licenceLength: '12M',
-          licensee: {
-            ...addressAndContact,
-            postalFulfilment: false,
-            preferredMethodOfConfirmation: 'Text',
-            preferredMethodOfReminder: 'Text',
-            mobilePhone: '07700900900',
-            preferredMethodOfNewsletter: 'Text'
-          }
-        }
-        const sampleRequest = generateRequestMock(permission)
-
-        const { summaryTable } = await getData(sampleRequest)
-
-        expect(summaryTable).toMatchSnapshot()
-      })
+    const sampleRequest = getRequestMock({
+      permission,
+      catalog: mssgCatalog,
+      setCurrentPermission: () => {}
     })
+    const { changeLicenceDetails } = await getData(sampleRequest)
+    expect(changeLicenceDetails).toBe(mssg)
+  })
 
-    describe('when purchasing a 1 day (non physical licence)', () => {
-      it('should display the Contact as the email and Newsletter as no', async () => {
-        const permission = {
-          licenceLength: '1D',
-          licensee: {
-            ...addressAndContact,
-            preferredMethodOfConfirmation: 'Email',
-            preferredMethodOfReminder: 'Email',
-            email: 'new3@example.com',
-            preferredMethodOfNewsletter: 'Prefer not to be contacted'
-          }
-        }
-        const sampleRequest = generateRequestMock(permission)
-
-        const { summaryTable } = await getData(sampleRequest)
-
-        expect(summaryTable).toMatchSnapshot()
+  it.each([
+    { preferredMethodOfNewsletter: HOW_CONTACTED.none, expectedReturn: mssgs.no, action: 'not sent' },
+    { preferredMethodOfNewsletter: null, expectedReturn: mssgs.yes, action: 'sent' }
```

Comment:
This test is really unclear. The title of the test is `'newsletter is $action when preferredMethodOfNewletter is $preferredMethodOfNewsletter'`, but in the test you don't assert on, and in fact don't even use `action`. Did you mean to?

If you are meaning to test on expectedReturn, what you've done isn't great practice. What you've done immediately above in `'changeLicenceDetails should be %s when isLicenceForYou is %s'` is much better, where you set the message in the catalog within the test, then assert that this is being used in the correct way in the function result.